client/my-sites/checkout/src/components/checkout-main.tsx

```diff
@@ -744,14 +744,16 @@ export default function CheckoutMain( {
 		[ responseCart.products ]
 	);
 
+	const product_slug_list = responseCart?.products?.map( ( product ) => product.product_slug );
```

Comment:
I love this idea. This diff made me realize that I don't actually know what we use this products data for in checkout anymore. We used to need it to add products to the cart, but that's no longer required since https://github.com/Automattic/wp-calypso/pull/59365. I think it may also sometimes have been used for the variant picker but that was made made obselete by https://github.com/Automattic/wp-calypso/pull/70452. The core checkout flow should no longer need the products data at all, although I believe I left this query component here because it's non-blocking and because I wasn't sure what other things might need it (eg: upsells?). 

Anyway, I think this change is fine for now, but we can probably remove many of these query components. I'll do that in another PR.

Reply:
Removing these in https://github.com/Automattic/wp-calypso/pull/87540